src/core/gui/XournalppCursor.cpp

```diff
@@ -515,6 +520,44 @@ auto XournalppCursor::createHighlighterOrPenCursor(int size, double alpha) -> Gd
 }
 
 
+auto XournalppCursor::createHorizontalLineCursor(int size, double alpha, GdkWindow* theWindow) -> GdkCursor* {
+    gint x, y;
+    GdkDevice *mouse_device;
+
+    #if GTK_CHECK_VERSION (3,20,0)
+    GdkSeat *seat = gdk_display_get_default_seat (gdk_display_get_default ());
+    mouse_device = gdk_seat_get_pointer (seat);
+    #else
+    GdkDeviceManager *devman = gdk_display_get_device_manager (gdk_display_get_default ());
+    mouse_device = gdk_device_manager_get_client_pointer (devman);
+    #endif
+
+    gdk_window_get_device_position (theWindow, mouse_device, &x, &y, NULL);
+
+    int height = size;
+    int width = gdk_window_get_width(theWindow);
+
+    int centerY = height / 2;
+    cairo_surface_t* crCursor = cairo_image_surface_create(CAIRO_FORMAT_ARGB32, width, height);
+    cairo_t* cr = cairo_create(crCursor);
+
+    cairo_set_line_width(cr, 1.2);
+    cairo_move_to(cr, 0, centerY);
+    cairo_line_to(cr, width, centerY);
+    cairo_close_path(cr);
+    cairo_stroke(cr);
+
+    cairo_destroy(cr);
+    GdkPixbuf* pixbuf = xoj_pixbuf_get_from_surface(crCursor, 0, 0, width, height);
```

Comment:
The same pixbuf is created over and over. Use some caching like for other cursors (using the `currentCursor` and `currentCursorFlavour` variables).

Reply:
Looks like I'm gonna have to add to an enum somewhere? Working on it...

Reply:
Y'know I'm not sure if caching is really gonna do much good here. The cursor needs to be re-drawn pretty much whenever
- the x-position of the mouse changes
- the position of the page changes
- the zoom level changes

There's not a whole lot I can do to "skip" steps, unless we want to save the pixbuf or something, but that seems like too much work for not a whole lot of reward. What do you think?

Reply:
My idea was to save the pixbuf. I don't know how big the reward is, though.

Reply:
I don't think saving the pixbuf is a worth it. I think any case we'd save that is a case where the new flavour value wouldn't change so we'd skip that step entirely.